### PR TITLE
fix: Speaker-Session not being linked

### LINKED ISSUE #1277: Speaker-Session not being linked
**Describe the bug**
When we create a new session, the session and the speaker are created in the database but aren't linked to each other. This is happening because of wrong post request. A manual postman request works fine on the server.

**To Reproduce**
1.Create a session
2. Speaker details aren't included

**Expected behavior**
The link should be formed properly

**Desktop (please complete the following information):**
 - OS: Ubuntu 18.04
 - Browser Chrome

### PR DESCRIPTION
**Pull request by mayank8318**
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

#### Short description of what this resolves:
Now the speaker and his corresponding session are linked together.

#### Changes proposed in this pull request:
- First save the speaker to the database so that it gets an ID.
- Now add this speaker to the session.speakers relationship
- Save both again

Fixes #1277

### CODE REVIEW

File: app/controllers/events/view/sessions/create.js [hunk 1]
```diff
@@ -3,13 +3,13 @@ export default Controller.extend({
   actions: {
     save() {
```
Comment (niranjan94) [inline app/controllers/events/view/sessions/create.js]: That's a lot of then & catch. Use `async` function and make this linear and more-readable.

**Comment by mayank8318** [comment]
@niranjan94 Made the changes

File: app/controllers/events/view/sessions/create.js [hunk 2]
```diff
@@ -1,27 +1,19 @@
 import Controller from '@ember/controller';
 export default Controller.extend({
   actions: {
-    save() {
+    async save() {
```
Comment (niranjan94) [inline app/controllers/events/view/sessions/create.js]: Make it 

```js
    async save() {
      try {
        this.set('isLoading', true);
        await this.get('model.speaker').save();
        this.get('model.speaker.sessions').pushObject(this.get('model.session'));
        await this.get('model.session').save();
        await this.get('model.speaker').save();
        this.get('notify').success(this.get('l10n').t('Your session has been saved'));
        this.transitionToRoute('events.view.sessions', this.get('model.event.id'));
      } catch (e) {
        this.get('notify').error(this.get('l10n').t('Oops something went wrong. Please try again'));
      }
    }
```

File: app/serializers/application.js [hunk 3]
```diff
@@ -21,7 +21,7 @@ export default JSONAPISerializer.extend(EventRelationMixin, {
   },
 
   serializeHasMany(snapshot, json, relationship) {
-    if (!snapshot.hasMany(relationship.key) || (snapshot.hasMany(relationship.key) && !snapshot.hasMany(relationship.key).id)) {
+    if (!snapshot.hasMany(relationship.key)) {
```
[the remark this answers is not in the document]
Comment (niranjan94) [reply app/serializers/application.js]: Can you log the result of `snapshot.hasMany(relationship.key)` (when there is data) and paste it here ?

**Comment by mayank8318** [comment]
@niranjan94 This is the output of `snapshot.hasMany(relationship.key)`
screenshot from 2018-06-21 15-16-28

**Comment by niranjan94** [comment]
@mayank8318 https://github.com/fossasia/open-event-frontend/pull/1291/files#diff-101f972450b2fc6ad1e826656cbf3d6eL24 is okay. Just make the other change.

**Comment by mayank8318** [comment]
@niranjan94 Made the changes

**Pull request merged**